lib/debug/repl.cpp

```diff
@@ -548,4 +541,71 @@ void Repl::addUartCommands(const std::uint8_t bus)
   }
 }
 
+void Repl::addAccelerometerCommands(const std::uint8_t bus, const std::uint8_t device_address)
+{
+  const auto optional_i2c = io::HardwareI2c::create(logger_, bus);
+  if (!optional_i2c) {
+    logger_.log(core::LogLevel::kFatal, "Failed to create I2C instance on bus %d", bus);
+    return;
+  }
+  const auto i2c = std::move(*optional_i2c);
+  const auto optional_accelerometer
+    = sensors::Accelerometer::create(logger_, i2c, bus, device_address);
+  const auto accelerometer = std::make_shared<sensors::Accelerometer>(*optional_accelerometer);
+  Command accelerometer_read_command;
+  std::stringstream identifier;
+  identifier << "accelerometer 0x" << std::hex << static_cast<int>(device_address) << " read";
+  accelerometer_read_command.name = identifier.str();
+  std::stringstream description;
+  description << "Read accelerometer sensor 0x" << std::hex << static_cast<int>(device_address)
+              << " on "
+              << "I2C bus " << static_cast<int>(bus);
+  accelerometer_read_command.description = description.str();
+  accelerometer_read_command.handler     = [this, accelerometer, bus]() {
+    const auto value = accelerometer->read();
+    if (!value) {
+      logger_.log(core::LogLevel::kFatal, "Failed to read the accelerometer from bus %d", bus);
+    } else {
+      const core::RawAccelerationData accelerometer_result = *value;
```

Comment:
`auto`

Reply:
i think we need to add this to our coding guide somewhere, if you can @TomLonergan03 could you add our actual coding practices in a wiki page? i agree with `auto`-ing optionals and enums, but not uints or other datatypes. Wdyt @kshxtij ?

Reply:
I agree, we definitely shouldn't auto uints cause @maxguy2001 would have a stroke

Reply:
Makes sense, last I heard `auto`  was for anything that was returned from a function but I could well be wrong.

And derefing optionals is kinda borderline for that still

Reply:
actually, there's quite a few edge cases, so it would be worth working out an explicit policy for `auto`

Reply:
dereferecing optionals is purely for speed but it does bypass exception checking (so we handle that). at this point I think it may be a good idea to have a HYPED style-guide, especially to help with next year @TomLonergan03

Reply:
using `auto` for derefing optionals is what i mean is borderline, i'll try to write something up but we should discuss it on monday

Reply:
though please do review my PR

Reply:
> I agree, we definitely shouldn't auto uints cause @maxguy2001 would have a stroke

CaN ThIs oNLy EvEr bE NeGaTivE???

Reply:
Yeah there seems to be a few areas where we disagree (including autos, initializing class members and where we store certain objects/definitions). While they don't usually impact the actual functionality of the  code too heavily, it's worth picking one method and sticking with it across the codebase for the sake of consistency.